### ISSUE TITLE
Feature Request: `@request.query.*` for realtime API

### ISSUE THREAD

**Issue by Lex-2008** [pocketbase/pocketbase#2514]
Hi,

Would it be possible to pass some extra query parameters to realtime API - in a similar way how it's currently possible to do it for simple records API - to be used in listRule?

Right now, I can create a collection "messages" with a field "key", and add a list API rule like this: `@request.query.key = key` - this way anonymous users can find all messages if they know a "key" for them, like this: `pb.collection('messages').getFullList({key:"..."})`.

However, this currently doesn't work for realtime API - it doesn't have a "query" option, so `@request.query.key` is always empty for them. In other words, realtime API works only for those records which you can see by issuing a simple `pb.collection('messages').getFullList()` call.

I was able to workaround it by creating a user for anyone accessing such "key-protected" records, recording all known keys into this user's "known" multi-relation field, and adding a listRule like this: `@request.auth.known.key?=key`. But I wonder if you might consider implementing it in the same way as it's done for records API. Thanks!

**Comment by ganigeorgiev** [comment]
I'll consider it together with the planned `expand` and `filter` params support (_most likely somewhere after the internal hooks refactoring and the JS app hooks integration_).

**Comment by Lex-2008** [comment]
Thanks!

**Comment by robitar** [comment]
Would it be possible to also specify the fields? If you just want to know that the collection changed, but plan to reconcile separately, you could request only the `id` field say, and avoid pulling down data you don’t need.

**Comment by ganigeorgiev** [comment]
Experimental `filter`, `expand`, `fields`, custom query and headers parameters support for the realtime subscription was added in the `develop` branch. There are some minor polishing details that needs to be done, but if you are curious you can test it with the [v0.20.0-rc prerelease](https://github.com/pocketbase/pocketbase/releases/tag/v0.20.0-rc). In general, it looks something like this:

```js
// JS SDK v0.20.0-rc
pb.collection("example").subscribe("*", (e) => {
  ...
}, {
  expand: "someRelField",
  filter: "status = 'active'",
  fields: "id,expand.someRelField.*:excerpt(100)",
})
```

**Comment by Lex-2008** [comment]
awesome! Now I have something to play with during the coming rainy weekend :)

**Comment by Lex-2008** [comment]
Sorry for a delay (was a bit busy at day job), tried it in my private project - works flawlessly for me! Both `expand`, `fields`, and extra parameters. Now my code can be quite shorter :) Thanks a lot!

**Comment by ganigeorgiev** [comment]
@mukhtharcm It is available in v0.20+. If you have troubles making it work, please file a [Q&A discussion](https://github.com/pocketbase/pocketbase/discussions/new?category=q-a) with more details about your use case and the problem you've stumbled on.

**Comment by mukhtharcm** [comment]
It's working!
I was confused about typescript was not showing suggestions :)
